src/pages/index.js

```diff
@@ -0,0 +1,293 @@
+
+  const name = editProfileForm.querySelector("#profile-name-input").value;
+  const about = editProfileForm.querySelector(
+    "#profile-description-input"
+  ).value;
+
+  api
+    .editUserInfo({ name, about })
+    .then((userData) => {
+      document.querySelector(".profile__name").textContent = userData.name;
+      document.querySelector(".profile__description").textContent =
+        userData.about;
+      closeModal(editModal);
+    })
+    .finally(() => showLoading(saveButton, false, "Saving...", "Save"));
+});
+
+// New event listener to pre-fill the form when opening the profile modal
+const editProfileButton = document.querySelector(".profile__edit-btn");
+editProfileButton.addEventListener("click", () => {
+  const editProfileForm = document.querySelector("#edit-profile-form");
+  const inputList = Array.from(
+    editProfileForm.querySelectorAll(settings.inputSelector)
+  );
+  // Clear validation errors and reset form state
+  resetValidation(editProfileForm, inputList, settings);
+  // Populate inputs with the current profile data
+  document.querySelector("#profile-name-input").value =
+    document.querySelector(".profile__name").textContent;
+  document.querySelector("#profile-description-input").value =
+    document.querySelector(".profile__description").textContent;
+
+  openModal(editModal); // Open the modal after setting the input values
+});
+
+// Avatar form submission
+avatarForm.addEventListener("submit", (evt) => {
+  evt.preventDefault();
+  const saveButton = avatarForm.querySelector(".modal__submit-btn");
+  showLoading(saveButton, true);
+  const avatarLink = avatarForm.querySelector("#profile-avatar-input").value;
+
+  api
+    .editAvatarInfo(avatarLink)
+    .then((userData) => {
+      document.querySelector(".profile__avatar").src = userData.avatar;
+      closeModal(avatarModal);
+    })
+    .finally(() => showLoading(saveButton, false));
+});
+
+// Add card form submission
+cardForm.addEventListener("submit", (evt) => {
+  evt.preventDefault();
+  const saveButton = cardForm.querySelector(".modal__submit-btn");
+  showLoading(saveButton, true, "Saving...", "Save");
+  const name = cardForm.querySelector("#card-caption-input").value;
+  const link = cardForm.querySelector("#add-card-link-input").value;
+
+  api
+    .addCard({ name, link })
+    .then((cardData) => {
+      // Add the new card to the cards list
+      document
+        .querySelector(".cards__list")
+        .prepend(createCardElement(cardData));
+
+      // Close the modal after adding
+      closeModal(cardModal);
+
+      // Reset the form fields and disable the button
+      cardForm.reset(); // Clears the input fields
+      disableButton(saveButton, settings); // Disable the save button
+
+      // Reset validation state for a clean slate
+      const inputList = Array.from(
+        cardForm.querySelectorAll(settings.inputSelector)
+      );
+      resetValidation(cardForm, inputList, settings);
```

Comment:
`resetValidation` is for clearing possible validation errors when you open the **profile** form, because you insert **valid** data into its inputs. But when you add a new card, you don't have any errors after submitting. 


Please, delete it